src/Snowflake/Command.php

```diff
@@ -140,6 +195,28 @@ public static function cloneDatabaseFromShared(Connection $connection, array $da
                 foreach ($tables as $table) {
                     $tableName = $table['name'];
 
+                    $tableGrants = array_filter(
+                        $tablesGrants,
+                        function (array $v) use ($database, $schemaName, $tableName) {
+                            $validSchema = [
+                                sprintf('%s.%s.%s', $database, $schemaName, $tableName),
+                                sprintf('%s.%s.%s', $database, $schemaName, QueryBuilder::quoteIdentifier($tableName)),
+                                sprintf('%s.%s.%s', $database, QueryBuilder::quoteIdentifier($schemaName), $tableName),
+                                sprintf(
+                                    '%s.%s.%s',
+                                    $database,
```

Comment:
předpoklad, že databáze nemají nikdy malá písmena v názvu je bohužel mylný :( třeba Sharedconfig se jmenuje `sapi_123` :(

Reply:
Tady by bylo fajn taky udělat vždycky nějaký "sanity assert" po tom filtru - třeba, že je tam aspoň jeden grant?

Reply:
> předpoklad, že databáze nemají nikdy malá písmena v názvu je bohužel mylný :( třeba Sharedconfig se jmenuje sapi_123 :(

tady ale jen hledám hodnotu v poli.. databáze zakládám oescapovaný, takže velikost písmen respektuju... tady je problém ten že mám schéma třeba `in.c-abcd` a tabulku `test`
no ale v tom poli je hodnota jako `"in.c-abcd".test` takže někdy to tam je s uvozovkami a někdy ne... a abych to nemusel řešit tak prostě vyzkouším všechny kombinace

u názvu DB jsem předpokládal že žádný speciální znaky nebudou takže to zkusím jen na hodnotu bez uvozovek

Reply:
> Tady by bylo fajn taky udělat vždycky nějaký "sanity assert" po tom filtru - třeba, že je tam aspoň jeden grant?

👍

Reply:
Speciální znaky ne, ale malá písmena ano (někde, historicky). 

`SAPI_123."in.c-abcd".test === SAPI_123."in.c-abcd".TEST`
`sapi_123."in.c-abcd".test === SAPI_123."in.c-abcd".TEST`
`"sapi_123"."in.c-abcd".test === "sapi_123"."in.c-abcd".TEST`

Reply:
jo to je pravda.... přidal jsem to i na databázi - https://github.com/keboola/project-migration-tool/pull/1/commits/6559e76589e446ddf6b70a2b9cd04d59c7c89342